src/utils/csv.rs

```diff
@@ -0,0 +1,26 @@
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use std::path::Path;
+
+pub fn load_data_from_csv<T: DeserializeOwned>(file_path: &Path) -> Result<Vec<T>, csv::Error> {
+    let mut csv_reader = csv::ReaderBuilder::new()
+        .has_headers(true)
+        .from_path(file_path)?;
+    let mut res = Vec::new();
+    for entry in csv_reader.deserialize() {
```

Comment:
nit: I think it should be possible to do something like `csv_reader.deserialize().collect::<Result<Vec<T>, _>>()?`

Reply:
I would love for it to, but sadly it is not iirc.